scripts/dts/extract_dts_includes.py

```diff
@@ -672,51 +672,62 @@ def print_key_value(k, v, tabstop):
     else:
         tabs = (len(label) >> 3) + 1
 
-    sys.stdout.write(label)
+    line = label
     for i in range(0, tabstop - tabs + 1):
-        sys.stdout.write('\t')
-    sys.stdout.write(str(v))
-    sys.stdout.write("\n")
+        line += '\t'
+    line += str(v)
+    line += '\n'
 
-    return
+    return line
 
 
-def generate_keyvalue_file(defs, args):
+def generate_keyvalue_file(defs, kv_file):
+    try:
+        fd = open(kv_file, "w")
+    except:
+        raise Exception(
+            "Input file " + os.path.abspath(kv_file) + " does not exist.")
 
     node_keys = sorted(defs.keys())
     for node in node_keys:
-        sys.stdout.write('# ' + node.split('/')[-1])
-        sys.stdout.write("\n")
+        fd.write('# ' + node.split('/')[-1])
+        fd.write("\n")
 
         prop_keys = sorted(defs[node].keys())
         for prop in prop_keys:
             if prop == 'aliases':
                 for entry in sorted(defs[node][prop]):
                     a = defs[node][prop].get(entry)
-                    sys.stdout.write("%s=%s\n" % (entry, defs[node].get(a)))
+                    fd.write("%s=%s\n" % (entry, defs[node].get(a)))
             else:
-                sys.stdout.write("%s=%s\n" % (prop, defs[node].get(prop)))
+                fd.write("%s=%s\n" % (prop, defs[node].get(prop)))
 
-        sys.stdout.write("\n")
+        fd.write("\n")
```

Comment:
Don't you have to close fd?

Perhaps you could use "with" instead of "try".

Reply:
it's run-once script and the files are opened only once, since all the fd are closed on exit I guess it's ok not to specifically close them.

Reply:
It's just good practice though, if I see this when reading code I have to read the rest of the program to verify that nothing else is opening the file at the same time.